apps/dashboard/app/models/batch_connect/session.rb

```diff
@@ -56,13 +56,33 @@ def get_binding
     # @return [Boolean] true if job is completed
     attr_accessor :cache_completed
 
+    # Error message about failing to parse info view ERB template.
+    # @return [String] error message
+    attr_reader :render_info_view_error_message
+
+    # Return parsed markdown from info.{md, html}.erb
+    # @return [String, nil] return HTML if no error while parsing, else return nil
+    def render_info_view
+      @render_info_view ||= OodAppkit.markdown.render(ERB.new(self.app.session_info_view, nil, "-").result(binding)).html_safe
```

Comment:
seems like instead of the guard clause for session_info_view in the rescue section you would do it here:

```
def render_info_view
        @render_info_view ||= OodAppkit.markdown.render(ERB.new(self.app.session_info_view, nil, "-").result(binding)).html_safe if self.app.session_info_view
```

Then if self.app.session_info_view is not set, render_info_view is nil, no error is raised render_info_view_error_message is nil and we handle that gracefully in the view.